### PR TITLE
Feature: Add Master Master replication

### PR DESCRIPTION
**Pull request by OliPou**

### CODE REVIEW

**Comment by ghost** [comment]
Thank you for your contribution.

File: README.md [hunk 1]
```diff
@@ -190,6 +190,22 @@ None.
         password: similarly-secure-password
         priv: "example_db.*:ALL"
 
+## Replication Master Master
```
Comment (tleguern) [inline README.md]: There are no sections dedicated to the Master/Slave replication mode so for consistency there should be either no section for Master/Master or an equivalent section for Master/Slave.

Comment (tleguern) [reply README.md]: Plus the order of words in the title is wrong: `Master Master Replication` seems better.

File: README.md [hunk 2]
```diff
@@ -190,6 +190,22 @@ None.
...
+    mysql_replication_master:
+    mysql_replication_role: [master01] / [master02]
+    mysql_server_id:
+
+to use it:
+
+    - hosts: Databases-Servers
+      become: yes
+      roles:
+        - { role: ansible-role-mysql, mysql_replication_master_master: true}
```
Comment (tleguern) [inline README.md]: Booleans are not necessarily the best idea, I would prefer `mysql_replication_mode: comaster|masterslave` for example. It is easier if other methods are added later and reduce the overall number of variables.

File: README.md [hunk 3]
```diff
@@ -190,6 +190,22 @@ None.
...
         priv: "example_db.*:ALL"
 
+## Replication Master Master
+
+Variable to define to use Master Master replication on each host:
+
+    mysql_logs_bin:
+    mysql_replication_master:
+    mysql_replication_role: [master01] / [master02]
+    mysql_server_id:
```
Comment (tleguern) [inline README.md]: Is this an example ? Why is only one variable assigned some values ? What does this value mean ?

File: tasks/replication-master-master.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
+---
+- name: Ensure replication user exists on every host.
+  mysql_user:
+    name: "{{ mysql_replication_user.name }}"
+    host: "{{ mysql_replication_user.host | default('%') }}"
+    password: "{{ mysql_replication_user.password }}"
+    priv: "{{ mysql_replication_user.priv | default('*.*:REPLICATION SLAVE,REPLICATION CLIENT') }}"
+    state: present
+  when:
+    - (mysql_replication_role == 'master01') or (mysql_replication_role == 'master02')
```
Comment (tleguern) [inline tasks/replication-master-master.yml]: Doesn't look good.

File: tasks/replication-master-master.yml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  mysql_replication:
+    mode: getslave
+  ignore_errors: true
+  register: slave
+  when:
+    - (mysql_replication_role == 'master01') or (mysql_replication_role == 'master02')
+    - (mysql_replication_master | length) > 0
+  tags: ['skip_ansible_galaxy']
+
+- name: Stop slave if error replication status is detected.
```
Comment (tleguern) [inline tasks/replication-master-master.yml]:
```suggestion
- name: Stop slave if replication status is in error.
```

File: tasks/replication-master-master.yml [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  register: restart_slave
+  ignore_errors: true
+  when:
+    - (slave.Slave_IO_Running is defined and slave.Slave_IO_Running != 'Yes') or (slave.Slave_IO_Running is defined and slave.Slave_SQL_Running != 'Yes')
+    - (mysql_replication_role == 'master01') or (mysql_replication_role == 'master02')
+    - (mysql_replication_master | length) > 0
+  tags: ['skip_ansible_galaxy']
+
+- name: Check master replication status.
+  mysql_replication: mode=getmaster
```
Comment (tleguern) [inline tasks/replication-master-master.yml]: This is not consistent with previous instances of `mysql_replication`.
```suggestion
  mysql_replication:
    mode: getmaster
```

File: tasks/replication-master-master.yml [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    master_log_pos: "{{ master.Position }}"
+  ignore_errors: true
+  when:
+    - (slave.Is_Slave is defined and not slave.Is_Slave) or (slave.Is_Slave is not defined and slave is failed) or (slave.Slave_IO_Running != 'Yes') or (slave.Slave_SQL_Running != 'Yes')
+    - (mysql_replication_role == 'master01') or (mysql_replication_role == 'master02')
+    - mysql_replication_user.name is defined
+    - (mysql_replication_master | length) > 0
+
+- name: Start replication.
+  mysql_replication: mode=startslave
```
Comment (tleguern) [inline tasks/replication-master-master.yml]: Same thing here.

File: tasks/replication-master-master.yml [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+- name: Check slave replication status.
+  mysql_replication:
+    mode: getslave
+  register: slave
+  when:
+    - (mysql_replication_role == 'master01') or (mysql_replication_role == 'master02')
+    - (mysql_replication_master | length) > 0
+  tags: ['skip_ansible_galaxy']
+
+- name: Generate error when slave status is failed
```
Comment (tleguern) [inline tasks/replication-master-master.yml]:
```suggestion
- name: Generate an error if slave status is failed
```

**Review by tleguern — CHANGES_REQUESTED** [review]

**Comment by OliPou** [comment]
I'll pull a new request with my new code

**Pull request closed without merging**